nncf/common/graph/graph.py

```diff
@@ -254,35 +337,92 @@ def _traverse_graph_recursive_helper(self, curr_node: NNCFNode,
                 self._traverse_graph_recursive_helper(node, traverse_function, output, traverse_forward)
         return output
 
-    def add_node(self, label: str, **attrs):
-        """
-        Adds node with `label` key (node_name) and `attr` attributes to the graph.
-
-        :param label: Key (node_name) of the node.
-        :param attrs: Attributes of the node.
-        """
-        node_id = len(self._node_id_to_key_dict)
-        attrs[NNCFGraph.ID_NODE_ATTR] = node_id
-        self._node_id_to_key_dict[node_id] = label
-        attrs[NNCFGraph.KEY_NODE_ATTR] = label
-        self._nx_graph.add_node(label, **attrs)
-
-        nncf_node = self.get_node_by_id(node_id)
-        if nncf_node.metatype in get_input_metatypes():
-            self._input_nncf_nodes[node_id] = nncf_node
-
-        if nncf_node.metatype in get_output_metatypes():
-            self._output_nncf_nodes[node_id] = nncf_node
-
-    def add_edge(self, u_of_edge: str, v_of_edge: str, **attrs):
-        """
-        Adds edge with attrs attributes between u and v.
-
-        :param u_of_edge: Producer node key (node_name).
-        :param v_of_edge: Consumer node key (node_name).
-        :param attrs: Attributes of the edge.
-        """
-        self._nx_graph.add_edge(u_of_edge, v_of_edge, **attrs)
+    def add_nncf_node(self, node_name: str,
+                      node_type: str,
+                      node_metatype: Type[OperatorMetatype],
+                      module_attributes: BaseLayerAttributes = None,
+                      node_id_override: int = None,
+                      layer_name: LayerName = None,
+                      ignored_algorithms: List[str] = None,
+                      is_in_iteration_scope: bool = False,
+                      is_integer_input: bool = False,
+                      is_shared: bool = False) -> NNCFNode:
+
```

Comment:
Please, add docstrings for this method and below.

Reply:
Done